crates/orchestrator/src/jobs/state_update_job/kzg.rs

```
@@ -0,0 +1,129 @@
+use crate::config::config;
+use crate::jobs::state_update_job::CURRENT_PATH;
+use alloy::eips::eip4844::BYTES_PER_BLOB;
+use c_kzg::{Blob, Bytes32, KzgCommitment, KzgProof, KzgSettings};
+use lazy_static::lazy_static;
+
+lazy_static! {
+    pub static ref KZG_SETTINGS: KzgSettings = KzgSettings::load_trusted_setup_file(
+        CURRENT_PATH.join("src/jobs/state_update_job/trusted_setup.txt").as_path()
+    )
+    .expect("Error loading trusted setup file");
+}
+
+/// Build KZG proof for a given block
+pub async fn build_kzg_proof(blob_data: Vec<u8>, x_0_value: Bytes32) -> color_eyre::Result<(Vec<u8>, KzgProof)> {
+    println!("blob_data: {:?}", blob_data.len());
+    let fixed_size_blob: [u8; BYTES_PER_BLOB] = blob_data.as_slice().try_into()?;
+
+    let blob = Blob::new(fixed_size_blob);
+    let commitment = KzgCommitment::blob_to_kzg_commitment(&blob, &KZG_SETTINGS)?;
+    let (kzg_proof, y_0_value) = KzgProof::compute_kzg_proof(&blob, &x_0_value, &KZG_SETTINGS)?;
+
+    // Verifying the proof for double check
+    let eval = KzgProof::verify_kzg_proof(
+        &commitment.to_bytes(),
+        &x_0_value,
+        &y_0_value,
+        &kzg_proof.to_bytes(),
+        &KZG_SETTINGS,
+    )?;
+    assert!(eval, "ERROR : Assertion failed, not able to verify the proof.");
```

this will panic, we should return an error